src/components/Login.js

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react';
+import { Redirect } from "react-router-dom";
+import './Login.css';
+
+class Login extends Component {
+  constructor(props) {
+    super(props);
+    // logout, to reset status
+    this.props.onLogout();
+
+    this.state = {
+      email: '',
+      password: '',
+      submitted: false
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(e) {
+    const { name, value } = e.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    this.setState({ submitted: true });
+
+    // If not blank, we attempt login
+    if (this.state.email && this.state.password) {
+      this.props.onLogin(this.state);
+    }
+  }
+
+  render() {
+    const { email, password, submitted } = this.state;
+    return this.props.isAuthenticated ? (
+        <Redirect to="/dashboard" />
+      ) : (
+        <div className="col-md-6 col-md-offset-3">
+          <h2>Login</h2>
+          <form name="form" onSubmit={this.handleSubmit}>
+            <div className={'form-group' + (submitted && !email ? ' has-error' : '')}>
+              <label htmlFor="email">Email</label>
+              <input
+                type="text"
+                className="form-control"
+                name="email"
+                value={email}
+                onChange={this.handleChange}
+              />
+              {submitted && !email && (
+                <div className="help-block">Email is required</div>
+              )}
+            </div>
+            <div className={'form-group' + (submitted && !password ? ' has-error' : '')}>
```

Comment:
This is a strange looking className. What's going on here?

Reply:
I've changed it slightly for the signup page, so this will get refactored. But essentially what is happening here is a ternary if statement will add the class name "has-error" if the user submits the form and hasn't filled out a password. We can then use that "has-error" class to add styling to indicate the error to the user.

Reply:
Ahh I see, okay. I think it would be a good idea to abstract this logic out of the div and into a helper method. Even with an understanding of what's going on, it's difficult to make sense of what is happening from just looking at it.

Reply:
[Here's](https://stackoverflow.com/questions/36771017/what-is-the-equivalent-to-ng-if-in-react-js) an approach I found on Stackoverflow that is accomplishing something similar. Ironically, it's an attempt at imitating behaviour that is native to Angular. Evidently we can't just have it all... :'(

```
return (
    this.props.showMe ? <button type="submit" className="btn nav-btn-red">SIGN UP</button> : null
  );
```